### PR TITLE
BUG Handle n_jobs=None in effective_n_jobs

### LINKED ISSUE #8: TypeError: '>' not supported between instances of 'NoneType' and 'int'
code:

```python
from sklearn.utils import parallel_backend
from sklearn.model_selection import cross_val_score
from sklearn import datasets
from sklearn import svm
from joblibspark import register_spark
from sklearn.cluster import DBSCAN, dbscan_
import numpy as np


register_spark()

if __name__ == "__main__":

  register_spark()  # register spark backend

  X1, y1 = datasets.make_circles(n_samples=5000, factor=.6,
                                 noise=.05)
  X2, y2 = datasets.make_blobs(n_samples=1000, n_features=2, centers=[[1.2, 1.2]], cluster_std=[[.1]],
                               random_state=9)
  X = np.concatenate((X1, X2))
  with parallel_backend('spark', n_jobs=3):
    y_pred = DBSCAN().fit_predict(X)
```
exception:

```pytb
  File "E:\pycharmWorkSpace\news-hot-analyze\36_venv\lib\site-packages\sklearn\cluster\_dbscan.py", line 389, in fit_predict
    self.fit(X, sample_weight=sample_weight)
  File "E:\pycharmWorkSpace\news-hot-analyze\36_venv\lib\site-packages\sklearn\cluster\_dbscan.py", line 335, in fit
    return_distance=False)
  File "E:\pycharmWorkSpace\news-hot-analyze\36_venv\lib\site-packages\sklearn\neighbors\_base.py", line 961, in radius_neighbors
    n_jobs = effective_n_jobs(self.n_jobs)
  File "E:\pycharmWorkSpace\news-hot-analyze\36_venv\lib\site-packages\joblib\parallel.py", line 389, in effective_n_jobs
    return backend.effective_n_jobs(n_jobs=n_jobs)
  File "E:\pycharmWorkSpace\news-hot-analyze\36_venv\lib\site-packages\joblibspark\backend.py", line 92, in effective_n_jobs
    if n_jobs > max_num_concurrent_tasks:
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```

### PR DESCRIPTION
**Pull request by madhuracj**
closes #8

### CODE REVIEW

**Comment by glemaitre** [comment]
@madhuracj You probably want to have a non-regression test.

**Comment by madhuracj** [comment]
Thank you for reviewing. 
Tests added. I hope you are fine with extracting new _get_max_num_concurrent_tasks() method to facilitate mocking the spark's behaviour.

**Comment by madhuracj** [comment]
@glemaitre Anything else that needs to be done?

File: test/test_backend.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    backend = SparkDistributedBackend()
+    max_num_concurrent_tasks = 8
+    backend._get_max_num_concurrent_tasks = MagicMock(return_value=max_num_concurrent_tasks)
+
+    assert backend.effective_n_jobs(n_jobs=None) == 1
+    assert backend.effective_n_jobs(n_jobs=-1) == 8
+    assert backend.effective_n_jobs(n_jobs=4) == 4
+
+    with warnings.catch_warnings(record=True) as w:
```
Comment (glemaitre) [inline test/test_backend.py]: depending on the type of warning, it would be better to use `pytest.mark.filterwarning` instead to decorate the test.

Comment (glemaitre) [reply test/test_backend.py]: Oh sorry I see that you are checking that a warning is raised. You want probably check that the content is what you are expecting

File: test/test_backend.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+from joblibspark.backend import SparkDistributedBackend
+
+
+def test_effective_n_jobs():
+
+    backend = SparkDistributedBackend()
+    max_num_concurrent_tasks = 8
+    backend._get_max_num_concurrent_tasks = MagicMock(return_value=max_num_concurrent_tasks)
+
+    assert backend.effective_n_jobs(n_jobs=None) == 1
```
Comment (glemaitre) [inline test/test_backend.py]: I would also check with the default parameter:

```python
assert backend.effective_n_jobs() == 8
```

**Comment by glemaitre** [comment]
@madhuracj I am not a contributor to the package so I will not merge the fix. 

ping @WeichenXu123

**Comment by WeichenXu123** [comment]
Good work, thanks!

**Pull request merged**